ui/src/app/shared/conditions-panel.tsx

```diff
@@ -26,6 +26,20 @@ export function hasWarningConditionBadge(conditions: Condition[]): boolean {
     return false;
 }
 
+export function hasArtifactGCError(conditions: Condition[]): boolean {
+    if (conditions.length === 0) {
+        return false;
+    }
+
```

Comment:
this "if" statement could probably be removed, right? (since it would be taken care of below)

Reply:
hmm, I'd be surprised but could be a peculiarity of the language I suppose

Reply:
I think the for-loop will cover the case when conditions is nil or if its length is 0. Check out [Go spec on RangeClause](https://go.dev/ref/spec#RangeClause). In addition, a simple "hello world" test can confirm the case.

Reply:
According to this, a nil slice would cause 0 iterations, therefore we should be able to remove this check in both functions. Testing now.

Reply:
@qhuai You may be right, but this is Typescript actually, not Go. :)